pxc/jenkins/pxc-package-testing-test-yash.groovy

```diff
@@ -0,0 +1,310 @@
+library changelog: false, identifier: 'lib@master', retriever: modernSCM([
+    $class: 'GitSCMSource',
+    remote: 'https://github.com/panchal-yash/jenkins-pipelines.git'
+]) _
+
+void installDependencies() {
+    sh '''
+        export PATH=${PATH}:~/.local/bin
+        sudo yum install -y git python3-pip jq
+        sudo amazon-linux-extras install ansible2
+        python3 -m venv venv
+        source venv/bin/activate
+        python3 -m pip install setuptools wheel
+        python3 -m pip install molecule==2.22 boto boto3 paramiko
+    '''
+    
+    sh '''
+        rm -rf package-testing
+        git clone https://github.com/panchal-yash/package-testing --branch wip-pxc-package-testing
+    '''
+
+    sh '''
+
+        echo "Private IP"
+        curl http://169.254.169.254/latest/meta-data/local-ipv4
+
+        echo "Public IP"
+        curl http://169.254.169.254/latest/meta-data/public-ipv4
+
+    '''
+
+}
+
+void runMoleculeAction(String action, String version, String scenario) {
+    def awsCredentials = [
+        sshUserPrivateKey(
+            credentialsId: 'MOLECULE_AWS_PRIVATE_KEY',
+            keyFileVariable: 'MOLECULE_AWS_PRIVATE_KEY',
+            passphraseVariable: '',
+            usernameVariable: ''
+        ),
+        aws(
+            accessKeyVariable: 'AWS_ACCESS_KEY_ID',
+            credentialsId: '5d78d9c7-2188-4b16-8e31-4d5782c6ceaa',
+            secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'
+        )
+    ]
+
+    withCredentials(awsCredentials) {
+        sh """
+            source venv/bin/activate
+            export MOLECULE_DEBUG=0
+            export install_repo=${params.INSTALL_REPO}
+            cd package-testing/molecule/pxc
+
+            cd ${version}-bootstrap
+            export INSTANCE_PRIVATE_IP=\${BOOTSTRAP_INSTANCE_PRIVATE_IP}
+            export INSTANCE_PUBLIC_IP=\${BOOTSTRAP_INSTANCE_PUBLIC_IP}            
+            molecule ${action} -s ${scenario}
+            cd -
+
+            cd ${version}-common
+            export INSTANCE_PRIVATE_IP=\${COMMON_INSTANCE_PRIVATE_IP}
+            export INSTANCE_PUBLIC_IP=\${COMMON_INSTANCE_PUBLIC_IP}        
+            molecule ${action} -s ${scenario}
+            cd -
+        """
+    }
+}
+
+void setInventories(){
+
+        sh """
+
+            echo \"Setting up Key path based on the selection\"
+
+            if  [[ ${params.TEST_DIST} == "ubuntu-focal" ]]
```

Comment:
The` if else ` part can be optimized per user. So if params.TEST_DIST in ubuntu, use ubuntu user and use `params.TEST_DIST` var in the path, if in debians, use admin etc.